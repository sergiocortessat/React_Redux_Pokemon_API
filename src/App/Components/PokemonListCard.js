import React from 'react';
import PropTypes from 'prop-types';
import {
  Link,
} from 'react-router-dom';
import '../../Styles/PokemonListCard.scss';

const CardListComponent = ({
  pokemon,
}) => (
  <Link to={`/stats/${pokemon.id}`} className="pokemon-list-link" data-testid="list">
    <div className="pokemon-list-card">
      <div>
        <h2>{`ID: ${pokemon.id}`}</h2>
        <h3>{`NAME(JP): ${pokemon.japanese}`}</h3>
        <h3>{`NAME(EN): ${pokemon.english}`}</h3>
        <div className="pokemon-list-card-image-container">
          <img src={pokemon.thumbnail} alt={pokemon.english} />
        </div>
        <h4>{`TYPE: ${pokemon.type[0]} - ${pokemon.type[1]}`}</h4>
        <h4>{`SPECIES: ${pokemon.species}`}</h4>
      </div>
    </div>
  </Link>
);
// default props
CardListComponent.defaultProps = {
  pokemon: {
    id: 0,
    japanese: '',
    english: '',
    type: ['', ''],
    species: '',
    thumbnail: '',
  },
};
CardListComponent.propTypes = {
  pokemon: PropTypes.shape({
    id: PropTypes.number,
    japanese: PropTypes.string,
    english: PropTypes.string,
    type: PropTypes.arrayOf(PropTypes.string),
    species: PropTypes.string,
    thumbnail: PropTypes.string,
  }),
};

export default CardListComponent;
