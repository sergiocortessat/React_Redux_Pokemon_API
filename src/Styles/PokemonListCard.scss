.pokemon-list-items {
  width: 100%;
  height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;

  .pokemon-list-link {
    text-decoration: none !important;
    color: inherit !important;
    margin: 10px 100px !important;

    .pokemon-list-card {
      background-color: salmon;
      border: 1px solid black;
      border-radius: 5% 20%;
      font-size: 1.7em;
      width: 320px;
      position: relative;

      > div {
        margin: 2px 3px 5px 30px;
      }
    }
  }
}
