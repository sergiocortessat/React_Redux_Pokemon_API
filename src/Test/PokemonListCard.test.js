import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import PokemonListCard from '../App/Components/PokemonListCard';

describe('Pokemon List Card', () => {
  test('should match with snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <PokemonListCard />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
