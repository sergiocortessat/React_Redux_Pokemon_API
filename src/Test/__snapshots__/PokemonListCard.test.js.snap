// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon List Card should match with snapshot 1`] = `
<a
  className="pokemon-list-link"
  data-testid="list"
  href="/stats/0"
  onClick={[Function]}
>
  <div
    className="pokemon-list-card"
  >
    <div>
      <h2>
        ID: 0
      </h2>
      <h3>
        NAME(JP): 
      </h3>
      <h3>
        NAME(EN): 
      </h3>
      <div
        className="pokemon-list-card-image-container"
      >
        <img
          alt=""
          src=""
        />
      </div>
      <h4>
        TYPE:  - 
      </h4>
      <h4>
        SPECIES: 
      </h4>
    </div>
  </div>
</a>
`;

exports[`Pokemon List Card with Link should match with snapshot 1`] = `
<a
  className="pokemon-list-link"
  data-testid="list"
  href="/stats/0"
  onClick={[Function]}
>
  <div
    className="pokemon-list-card"
  >
    <div>
      <h2>
        ID: 0
      </h2>
      <h3>
        NAME(JP): 
      </h3>
      <h3>
        NAME(EN): 
      </h3>
      <div
        className="pokemon-list-card-image-container"
      >
        <img
          alt=""
          src=""
        />
      </div>
      <h4>
        TYPE:  - 
      </h4>
      <h4>
        SPECIES: 
      </h4>
    </div>
  </div>
</a>
`;
